### PR TITLE
RP-612 TenantAdmin - add id & key not null to validation

### PR DESCRIPTION
**Pull request by vgonzalez9**
re-add the validation on the tenant/sandbox keys & id are not null or empty.
added tests.

### CODE REVIEW

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationService.java [hunk 1]
```diff
@@ -84,6 +84,12 @@ void verifySandboxRequiredFields(TenantConfiguration tenantConfiguration) {
      */
     private void verifyNotExists(final Tenant tenant, final Map<String, TenantConfiguration> knownTenantConfigurations) {
         String tenantType = tenant.isSandbox() ? "Sandox" : "Tenant";
+        if (tenant.getKey() == null || tenant.getKey().isEmpty() || tenant.getId() == null || tenant.getId().isEmpty()) {
```
Comment (ftdholbrook) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/service/impl/DefaultTenantValidationService.java]: I think this method is doing a bit too much.  maybe create a separate method for required field checks?  Something similar to verifySandboxRequiredFields

File: admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/database/impl/RdwDatabaseType.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
     REPORTING("reporting_r", false),
     WAREHOUSE("warehouse_r", false);
 
-    final String datasourceNameHint;
+    public final String datasourceNameHint;
```
Comment (esotrope) [inline admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/database/impl/RdwDatabaseType.java]: should use a getter no? this makes it mutable

Comment (ftdholbrook) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/database/impl/RdwDatabaseType.java]: it is an enum, and final, can't be mutated

Comment (esotrope) [reply admin-service/src/main/java/org/opentestsystem/rdw/admin/multitenant/database/impl/RdwDatabaseType.java]: ah good point. seeing the public just pushed my frazzled mind over the edge x J

**Pull request merged**